src/csharp/Microsoft.Spark.Worker/Processor/CommandProcessor.cs

```diff
@@ -100,6 +100,7 @@ private static RDDCommand ReadRDDCommand(Stream stream)
                 (2, 3) => SqlCommandProcessorV2_3_X.Process(evalType, stream),
                 (2, 4) => SqlCommandProcessorV2_4_X.Process(evalType, stream),
                 (3, 0) => SqlCommandProcessorV2_4_X.Process(evalType, stream),
+                (3, 1) => SqlCommandProcessorV2_4_X.Process(evalType, stream),
```

Comment:
Can we do `(3, _)` so that this doesn't force us to do a major version bump for a Spark minor version bump?

Reply:
We definitely can.  If we want to be more lenient and remove these artificial version checking gates, should we also revisit the version check we've been adding to the `jar`s ?  In the jar, instead of validating against supported versions, do the reverse and check against unsupported versions, this way users won't have to update the jar every time a patch version is released and they want to use the same jar.

Reply:
JARs seem fine to me since they don't affect the backward compatibility. Since patch releases are >6 months apart anyway, I am fine with releasing a new version after the verification is done. But, I am open to this idea if there are strong reasons.

Reply:
https://github.com/dotnet/spark/discussions/846

I think we can introduce a config to bypass the check? The user can use the config on his/her own risk. WDYT?

Reply:
I'm okay with an opt-in approach to bypassing the checks.  Is this only in regards to the jar or are there other places you had in mind (like in checks modified in this PR).

Reply:
Just JARs should be enough for now.

Reply:
BTW, this will only bypass checks on the Spark's patch version. We still need to check the minor version.